### PR TITLE
Add unit tests for actionUtils

### PR DESCRIPTION
**Pull request by joshmgross**
Following up from https://github.com/actions/cache/pull/62

Adding unit tests for utilities in this action (`actionUtils`)

### CODE REVIEW

File: __tests__/actionUtils.test.ts [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+});
+
+test("isExactKeyMatch with undefined cache entry returns false", () => {
+    const key = "linux-rust";
+    const cacheEntry = undefined;
+
+    expect(actionUtils.isExactKeyMatch(key, cacheEntry)).toBe(false);
+});
+
+test("isExactKeyMatch with undefined cache entry key returns false", () => {
```
Comment (yacaovsnc) [inline __tests__/actionUtils.test.ts]: ...with `empty` cache entry key...

File: __tests__/actionUtils.test.ts [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+    const homedir = jest.requireActual("os").homedir();
+    const homedirMock = jest.spyOn(os, "homedir");
+    homedirMock.mockImplementation(() => {
+        return homedir;
+    });
+
+    const resolvedPath = actionUtils.resolvePath(filePath);
+
+    const expectedPath = path.join(homedir, ".cache/yarn");
```
Comment (yacaovsnc) [inline __tests__/actionUtils.test.ts]: Does this work on windows?  `/` isn't replaced with `\`?

Comment (joshmgross) [reply __tests__/actionUtils.test.ts]: This will replace `/` with `\` on Windows, which matches the behavior of `actionUtils.resolvePath`

Comment (yacaovsnc) [reply __tests__/actionUtils.test.ts]: Oh oh, I missed it's `path` join.

**Pull request merged**